src/main/java/com/blamejared/crafttweaker/impl_native/util/ExpandVector3d.java

```diff
@@ -0,0 +1,139 @@
+                @NativeConstructor.ConstructorParameter(type = double.class, name = "z")
+        })
+)
+public class ExpandVector3d {
+    @ZenCodeType.Getter("x")
+    public static double getX(final Vector3d $this) {
+        return $this.getX();
+    }
+
+    @ZenCodeType.Getter("y")
+    public static double getY(final Vector3d $this) {
+        return $this.getY();
+    }
+
+    @ZenCodeType.Getter("z")
+    public static double getZ(final Vector3d $this) {
+        return $this.getZ();
+    }
+
+    @ZenCodeType.Method("subtractReverse")
+    public static Vector3d subtractReverse(final Vector3d $this, final Vector3d other) {
+        return $this.subtractReverse(other);
+    }
+
+    @ZenCodeType.Method("normalize")
+    public static Vector3d normalize(final Vector3d $this) {
+        return $this.normalize();
+    }
+
+    @ZenCodeType.Method("dot")
+    public static double dot(final Vector3d $this, final Vector3d other) {
+        return $this.dotProduct(other);
+    }
+
+    @ZenCodeType.Method("cross")
+    public static Vector3d cross(final Vector3d $this, final Vector3d other) {
+        return $this.crossProduct(other);
+    }
+
+    @ZenCodeType.Method("times")
+    @ZenCodeType.Operator(ZenCodeType.OperatorType.MUL)
+    public static Vector3d mul(final Vector3d $this, final Vector3d other) {
+        return $this.mul(other);
+    }
+
+    @ZenCodeType.Method("times")
+    public static Vector3d mul(final Vector3d $this, final double x, final double y, final double z) {
+        return $this.mul(x, y, z);
+    }
+
+    @ZenCodeType.Method("add")
+    @ZenCodeType.Operator(ZenCodeType.OperatorType.ADD)
+    public static Vector3d add(final Vector3d $this, final Vector3d other) {
+        return $this.add(other);
+    }
+
+    @ZenCodeType.Method("add")
+    public static Vector3d add(final Vector3d $this, final double x, final double y, final double z) {
+        return $this.add(x, y, z);
+    }
+
+    @ZenCodeType.Method("subtract")
+    @ZenCodeType.Operator(ZenCodeType.OperatorType.SUB)
+    public static Vector3d sub(final Vector3d $this, final Vector3d other) {
+        return $this.subtract(other);
+    }
+
+    @ZenCodeType.Method("subtract")
+    public static Vector3d sub(final Vector3d $this, final double x, final double y, final double z) {
+        return $this.subtract(x, y, z);
+    }
+
+    @ZenCodeType.Method("getCoordinate")
+    @ZenCodeType.Operator(ZenCodeType.OperatorType.INDEXGET)
+    public static double getCoordinate(final Vector3d $this, final int index) {
+        switch (index) {
+            case 0: return $this.getX();
+            case 1: return $this.getY();
+            case 2: return $this.getZ();
+        }
+        throw new ArrayIndexOutOfBoundsException(index);
+    }
+
+    @ZenCodeType.Method("getCoordinate")
+    @ZenCodeType.Operator(ZenCodeType.OperatorType.INDEXGET)
+    public static double getCoordinate(final Vector3d $this, final Direction.Axis axis) {
```

Comment:
Have you tested this?
I'm not sure how well ZC currently handles overloaded operators